src/Common.Lib/Macros.h

```diff
@@ -22,15 +22,21 @@
 #define __FUNCTIONT__
 #endif
 
+// Asserts and optionally logs that a failure occured with the given hresult.
+// Note, this library does not define the implementation of this function. It
+// is up to the binary executable that links this library to supply a definition
+// otherwise, linkage errors will be expected.
+void AssertLogFailure(_In_ const WCHAR* wszError, ...);
+
     // Log an error without returning.
 #ifndef IfFailLog
 #define IfFailLog(EXPR) \
-    do { if (FAILED(hr = (EXPR))) { CLogging::LogError(_T("IfFailLog(") _T(#EXPR) _T(") failed in function ") __FUNCTIONT__ _T(": %d"), hr); } } while (false)
```

Comment:
Are we actually calling any of these in Common.lib (i'm not sure how this could have worked if we were)? Maybe we can move this to common.headers and skip the intermediate AssertLogFailure function.

Reply:
This could be moved to Common.Headers regardless, honestly. I left it here because it isn't technically a header-only library. You need to supply an implementation of AssertLogFailure.

Honestly, I mostly did this so that the header can be reused inside Nagler. But Nagler won't have access to CLogging, so I needed a work-around. Hence, AssertLogFailure. I could not use this in Nagler, but it would mean duplicating the dll exports, etc.




---
In reply to: [573096085](https://github.com/microsoft/CLRInstrumentationEngine/pull/362#discussion_r573096085) [](ancestors = 573096085)